src/app/workflow/executions/executions-tab.component.ts

```diff
@@ -0,0 +1,218 @@
+    this.extendedGA4GHService
+      .aggregatedMetricsGet(this.trsID, this.version.name)
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe(
+        (metrics) => {
+          this.resetMetricsData();
+          if (metrics) {
+            for (const [partner, metric] of Object.entries(metrics)) {
+              this.setMetricsObject(this.metrics, partner, metric);
+            }
+            this.partners = Array.from(this.metrics.keys());
+            if (this.partners.length > 0) {
+              this.metricsExist = true;
+              this.selectPartner(this.partners[0]);
+            }
+          }
+          this.alertService.detailedSuccess();
+        },
+        (error) => {
+          this.alertService.detailedError(error);
+        }
+      );
+  }
+
+  private setMetricsObject(metricsMap: Map<PartnerEnum, Metrics>, partner: string, metric: Metrics) {
+    for (const partnerEnum in PartnerEnum) {
+      if (PartnerEnum[partnerEnum] === partner) {
+        metricsMap.set(PartnerEnum[partnerEnum], metric);
+      }
+    }
+  }
+
+  private resetMetricsData() {
+    this.metricsExist = false;
+    this.metrics = new Map();
+    this.metricsTable = null;
+    this.validationsTable = null;
+    this.partners = [];
+    this.totalExecutions = null;
+    this.successfulExecutions = null;
+    this.failedExecutions = null;
+    this.currentPartner = null;
+    this.metricsColumns = null;
+    this.validationsColumns = null;
+  }
+
+  private loadMetricsData(partner: PartnerEnum) {
+    this.metricsTable = [];
+    const metrics = this.metrics.get(partner);
+    if (metrics) {
+      this.metricsColumns = ['metric', 'min', 'avg', 'max'];
+      this.insertToMetricsTable(this.metricsTable, metrics?.cpu, 'CPU');
+      this.insertToMetricsTable(this.metricsTable, metrics?.memory, 'Memory');
+      this.insertToMetricsTable(this.metricsTable, metrics?.executionTime, 'Run Time');
+      if (metrics.executionStatusCount) {
+        this.totalExecutions =
+          metrics.executionStatusCount.numberOfSuccessfulExecutions + metrics.executionStatusCount.numberOfFailedExecutions;
+        this.successfulExecutions = metrics.executionStatusCount.numberOfSuccessfulExecutions;
+        this.failedExecutions = metrics.executionStatusCount.numberOfFailedExecutions;
+      }
+    }
+  }
+
+  private insertToMetricsTable(table: any[], metricsData: CpuMetric | MemoryMetric | ExecutionTimeMetric, metricLabel: string) {
+    table.push({
+      metric: metricLabel,
+      min: metricsData?.minimum,
+      avg: metricsData?.average,
+      max: metricsData?.maximum,
+      unit: metricsData?.unit,
+    });
+  }
+
+  private loadValidationsData(partner: PartnerEnum) {
+    const validations = this.metrics.get(partner).validationStatus?.validatorToolToIsValid;
+    if (validations) {
+      for (const [validatorTool, isValid] of Object.entries(validations)) {
+        this.validationsTable = [
```

Comment:
I don't know the data structure of the validations very well, but is this intentional? Because you're in a loop, the second validation, if there is one, is going to overwrite `this.validationsTable`, isn't it?

Reply:
Belated thought, is the idea that you just want to display the last validation only? If so, it would be less confusing to have no loop and just process the last element.

Reply:
Oops, sorry. I meant to push into `this.validationsTable` instead of overwriting. Thanks for catching that!

Reply:
Since there may be more than one validator in the table, I changed the "Most recent error message" column to always be present. If a validator tool's most recent run is valid, that column would just be blank for that row.

Reply:
Screenshots in description is now updated